### PR TITLE
feat(custom_op): use dynamic process number

### PR DESCRIPTION
**Pull request by JieRen98**
## Description

use dynamic process number instead of the fixed number 32

### CODE REVIEW

File: src/adam_op/adam_op_impl.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ void adamForwardInplaceCPUKernel(
     const other_t inv_one_minus_pow_b2, const other_t eps,
     const other_t eps_root, const size_t n, scalar_t* __restrict__ updates_ptr,
     scalar_t* __restrict__ mu_ptr, scalar_t* __restrict__ nu_ptr) {
-#pragma omp parallel for num_threads(32)
+#pragma omp parallel for num_threads(omp_get_num_procs())
```
Comment (XuehaiPan) [inline src/adam_op/adam_op_impl.cpp:34]: Is this machine dependent? We are building pypi wheels in GitHub Action containers.

Comment (JieRen98) [reply src/adam_op/adam_op_impl.cpp:34]: no, depending on the user's machine (runtime decided)

**Comment by XuehaiPan** [comment]
Please update the CHANGELOG.md. Then we can trigger a merge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Use dynamic process number in CPU kernels by [@JieRen98](https://github.com/JieRen98) in [#42](https://github.com/metaopt/TorchOpt/pull/42).
```
Comment (XuehaiPan) [inline CHANGELOG.md:18]: The v0.4.2 have already been tagged. Please add this line to the "Unreleased" section.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ------
 
 ## [Unreleased]
-
+- Use dynamic process number in CPU kernels by [@JieRen98](https://github.com/JieRen98) in [#42](https://github.com/metaopt/TorchOpt/pull/42).
```
Comment (XuehaiPan) [inline CHANGELOG.md:11]:
```suggestion

### Added

Use dynamic process number in CPU kernels by [@JieRen98](https://github.com/JieRen98) in [#42](https://github.com/metaopt/TorchOpt/pull/42).

```

**Review by XuehaiPan — APPROVED** [review]

**Pull request merged**